app/controllers/passwords_controller.rb

```diff
@@ -34,7 +33,8 @@ def update
     user = User.find_by(token: params[:token])
     if user
       flash[:success] = "Your password has been changed. Please log in."
-      user.update_attributes(password: params[:password], token: nil)
+      user.update_attribute(:password, params[:password])
```

Comment:
I originally did this because for some reason `update_attributes` was given me trouble. More specifically, I use `before_validation` for creating a token, and that was giving me trouble. Though, you're right, it seems to work fine now.